src/modules/flow/netctl/netctl.json

```diff
@@ -0,0 +1,139 @@
+{
+  "$schema": "http://solettaproject.github.io/soletta/schemas/node-type-genspec.schema",
+  "name": "netctl",
+  "meta": {
+    "author": "Intel Corporation",
+    "license": "Apache-2.0",
+    "version": "1"
+  },
+  "types": [
+    {
+      "category": "netctl",
+      "description": "Send list of all network services attached to board.",
+      "methods": {
+        "open": "open_list_services",
+        "close": "close_list_services"
+      },
+      "name": "netctl/list-services",
+      "out_ports": [
+        {
+          "data_type": "string",
+          "description": "Name of network services supported by system.",
+          "name": "OUT"
+        }
+      ],
+      "private_data_type": "list_service_data"
+    },
+    {
+      "category": "netctl",
+      "description": "Connect or disconnect to one available network access point.",
+      "in_ports": [
+        {
+          "data_type": "any",
+          "description": "Trigger to connect to network service.",
+          "methods": {
+            "process": "connect_service_name"
+          },
+          "name": "CONNECT"
+        },
+        {
+          "data_type": "any",
+          "description": "Trigger to disconnect to network service.",
+          "methods": {
+            "process": "disconnect_service_name"
+          },
+          "name": "DISCONNECT"
+        }
+      ],
+      "methods": {
+        "open": "open_service_connection",
+        "close": "close_service_connection"
+      },
+      "name": "netctl/service",
+      "options": {
+        "members": [
+          {
+            "data_type": "string",
+            "default": null,
+            "description": "Refer to netctl/list-services on how to get the service name",
+            "name": "name"
+          }
+        ],
+        "version": 1
+      },
+      "out_ports": [
+        {
+          "data_type": "string",
+          "description": "Output network services connection status.",
+          "name": "OUT"
```

Comment:
Here I'd have an extra output port `ONLINE`, a boolean that checks if the `OUT` is online. Then it's simple to use and most people will just use that one. Providing a `OUT` will all states will be useful for some specific applications.